File: VEcchi/Python/Fatti/Scomposizione.py
j = input("Inserire il numero da scomporre in fattori primi: ")

def isPrime(a):
    div = a - 1
    res = True

    while(div > 1):
        if a % div == 0:
            res = False
        div = div - 1

    return res

def scomposizione(a):
    if a > 1:
        p = 2
        
        while(a > 1):
            if isPrime(p):
                if a % p == 0:
                    print p
                    a = a / p
                else:
                    p = p + 1
            else:
                p = p + 1
    else:
        print "Inserire un numero maggiore di 1."

scomposizione(j)

File: Scomposizione/src/Scomposizione.java
import java.util.Scanner;


public class Scomposizione {
	private static Scanner console;
	private static boolean res;

	public static void main(String[] args){
		console = new Scanner(System.in);
		System.out.println("Inserisci un valore da scomporre: ");
		int a = console.nextInt();
		int p = 2;
		if (a > 1){
			while(a > 1){
				if(isPrime(a)){
					if(a % p==0){
						System.out.print(p + "\t");
						a /= p;
					}else{
						p += 1;
					}
				}else{
					p += 1;
				}
			}
		}else{
			System.out.print("Inserisci un numero maggiore a 1.");
		}
		
	}

	public static boolean isPrime(int n) {
		int div = n - 1;
		while(div > 1){
			if (n % div==0) return true;
			div--;
		}
		return true;
		
		
	}

	public static boolean isRes() {
		return res;
	}

	public static void setRes(boolean res) {
		Scomposizione.res = res;
	}
	
}